src/expressions/if-expr.md

```diff
@@ -4,11 +4,21 @@
 
 > **<sup>Syntax</sup>**\
 > _IfExpression_ :\
-> &nbsp;&nbsp; `if` [_Expression_]<sub>_except struct expression_</sub> [_BlockExpression_]\
-> &nbsp;&nbsp; (`else` (
+> &nbsp;&nbsp; `if` _inIfList_ [_BlockExpression_]\
+> &nbsp;&nbsp; ( `else` (
 >   [_BlockExpression_]
 > | _IfExpression_
 > | _IfLetExpression_ ) )<sup>\?</sup>
+>
+> _inIf_ :\
+> &nbsp;&nbsp; ( `let` [_Pattern_] `=` [_Expression_]<sub>_except struct expression_</sub>
+> | [_Expression_]<sub>_except struct expression_</sub>
+> | _inIf_ )
+>
+> _inIfList_ :\
+> &nbsp;&nbsp; _inIf_ ( && _inIf_ )*
+
+[_Expression_]<sub>_except struct expression_</sub>
```

Comment:
@ehuss Before jumping into `while` and `if let`, do you recommend some way to avoid duplication? `if`, `if let` and `while` will mostly have the same grammar

Reply:
Sorry, I'm not sure I understand the question.  

If I understand the RFC correctly, there no longer is a grammatical difference between `if` and `if let`.  What I would do is:

* Remove distinctions between `if` and `if let`:
     * Remove _IfLetExpression_ as a distinct rule. 
     * Change the name of the chapter to just be "`if` expressions"
     * Rewrite `if-expr.md` so that it doesn't discuss them as two separate kinds of expressions.
* Define _InIfList_ once in `if-expr.md`
* Perform the same treatment to `while let`, removing _PredicatePatternLoopExpression_, and joining those two sections.  In the rules, you can link to other pages, so `while` would become something like:
  > `while` _InIfList_ _BlockExpression_
  
  Where _InIfList_ links to `if-expr.md`

Reply:
Thanks @ehuss 

The PR was updated according to your indications